File: src/p020/p020.hs
module P020 where

import Data.Char (digitToInt)

factorial :: Integer -> Integer
factorial n = product [1 .. n]

factorialDigitSum :: Integer -> Int
factorialDigitSum n = sum $ map digitToInt $ show $ factorial n

main :: IO ()
main = do
  print $ factorialDigitSum 100

File: src/p020/p020.py
#!/usr/bin/env python3

from math import prod


def factorial_of(num):
    return prod(n for n in range(1, num))


def factorial_digit_sum(num):
    factorial = factorial_of(num)
    return sum(int(digit) for digit in str(factorial))


if __name__ == "__main__":
    print(factorial_digit_sum(100))
